drivers/Watchdog.cpp

```diff
@@ -18,48 +18,103 @@
 
 #include "Watchdog.h"
 
-namespace mbed {
+namespace mbed
+{
+
+Watchdog *Watchdog::first;
 
-watchdog_status_t Watchdog::start(const uint32_t timeout)
+bool Watchdog::add_to_list()
```

Comment:
If you use a flag to keep track of if an instance has been added to the list, such as `_is_initialized`, then you don't need to loop through the list. You can add this element immediately to the front of the list.